src/dev/adam_practice_tagfinder.py

```diff
@@ -0,0 +1,66 @@
+import bs4
+
+with open("sample_webpage.html") as html_file:
+    html = html_file.read()
+
+soup = bs4.BeautifulSoup(html, "html.parser")
+
+def p_tag_finder(my_soup):
+
+    # Lines 12 - 17 find all the <p> tags in the html file and count them, creating the...
+    # ...variable called 'total_number_of_p_tags'.
+    p_tags = my_soup.findAll('p')
+
+    total_number_of_p_tags = 0
+
+    for item in p_tags:
+        total_number_of_p_tags +=1
+
+    # Lines 21 - 24 find ALL parents of EVERY <p> tag and add them to the list...
+    # ...called 'p_tag_parents'.
+    p_tag_parents = []
+
+    for tag in p_tags:
```

Comment:
Better to use python list comprehensions whenever possible for things like this, I believe they are more efficient:

```suggestion
    p_tag_parents = [tag.parents for tag in p_tags]
```